src/components/ui/Badge/Badge.tsx

```diff
@@ -10,17 +12,41 @@ type Props = {
   small?: boolean
   variant?: BadgeVariants
   children: ReactNode
+  onClose?: () => void
+  interactive?: boolean
 }
```

Comment:
Also, we can try this:

-  if have interactive, onClose is not required
- if havent interactive, onClose is not permitted
- if have onClose, interactive is required

```suggestion
  small?: boolean
  variant?: BadgeVariants
  children: ReactNode
  onClose?: () => void
  interactive?: boolean
} & BadgeOnClose

type BadgeOnClose =
  | {
      interactive: true
      onClose?: () => void
    }
  | {
      interactive?: false
      onClose?: never
    }
```